reconcile/test/aws_account_manager/test_aws_account_manager_reconciler.py

```diff
@@ -0,0 +1,824 @@
+) -> None:
+    reconciler._tag_account(
+        aws_api, {"default": "tag"}, "account", "123456789012", {"new": "tag"}
+    )
+
+    aws_api.organizations.tag_resource.assert_called_once_with(
+        resource_id="123456789012",
+        tags={"default": "tag", "new": "tag"},
+    )
+    aws_api.organizations.untag_resource.assert_not_called()
+
+
+def test_aws_account_manager_reconcile_tag_account_state_exists_and_is_tagged(
+    aws_api: MagicMock, reconciler: AWSReconciler, state_exists: Callable
+) -> None:
+    state_exists(
+        state_key("account", TASK_TAG_ACCOUNT), {"default": "tag", "new": "tag"}
+    )
+    reconciler._tag_account(
+        aws_api, {"default": "tag"}, "account", "123456789012", {"new": "tag"}
+    )
+
+    aws_api.organizations.tag_resource.assert_not_called()
+    aws_api.organizations.untag_resource.assert_not_called()
+
+
+def test_aws_account_manager_reconcile_tag_account_state_exists_and_needs_update(
+    aws_api: MagicMock, reconciler: AWSReconciler, state_exists: Callable
+) -> None:
+    state_exists(
+        state_key("account", TASK_TAG_ACCOUNT), {"default": "tag", "foo": "bar"}
+    )
+    reconciler._tag_account(
+        aws_api, {"default": "tag"}, "account", "123456789012", {"new": "tag"}
+    )
+
+    aws_api.organizations.untag_resource(aws_api, ["default", "new"])
+    aws_api.organizations.tag_resource.assert_called_once_with(
+        resource_id="123456789012",
+        tags={"default": "tag", "new": "tag"},
+    )
+
+
+def test_aws_account_manager_reconcile_tag_account_dry_run(
+    aws_api: MagicMock, reconciler_dry_run: AWSReconciler
+) -> None:
+    reconciler_dry_run._tag_account(
+        aws_api, {"default": "tag"}, "account", "123456789012", {"new": "tag"}
+    )
+    aws_api.organizations.tag_resource.assert_not_called()
+    aws_api.organizations.untag_resource.assert_not_called()
+
+
+def test_aws_account_manager_reconcile_get_destination_ou(
+    aws_api: MagicMock, reconciler: AWSReconciler
+) -> None:
+    aws_api.organizations.get_organizational_units_tree.return_value = (
+        AwsOrganizationOU(
+            Id="ou-123456",
+            Arn="arn",
+            Name="Root",
+            children=[AwsOrganizationOU(Id="ou-123457", Arn="arn", Name="ou")],
+        )
+    )
+    ou = reconciler._get_destination_ou(aws_api, "/Root/ou")
+    assert ou.name == "ou"
+
+    with pytest.raises(KeyError):
+        reconciler._get_destination_ou(aws_api, "/not/found")
+
+
+def test_aws_account_manager_reconcile_move_account(
+    aws_api: MagicMock, reconciler: AWSReconciler
+) -> None:
+    reconciler._get_destination_ou = MagicMock(  # type: ignore
+        return_value=AwsOrganizationOU(Id="ou-123457", Arn="arn", Name="ou")
+    )
+
+    reconciler._move_account(aws_api, "account", "123456789012", "ou")
+    aws_api.organizations.move_account.assert_called_once_with(
+        uid="123456789012", destination_parent_id="ou-123457"
+    )
```

Comment:
partial mock a private method to test another private method seems unnatural, ideally should mock dependencies like `aws_api` behaviour, otherwise this test will become a blocker for refactor as it's coupled with implementation instead of behaviour.

It seems many test cases in this file are using this style too, typical when add tests after implementation is done.

Reply:
I know and I accept this technical dept and leave it to my future self ;) 
In this case, I don't want something like `get_destination_ou_func: Callable = _get_destination_ou`